components/layout_2020/table/layout.rs

```diff
@@ -63,24 +86,473 @@ impl<'a> TableLayout<'a> {
+        // > the smaller of:
+        // >   * the intrinsic percentage width of the column based on cells of span up to N,
+        // >     where N is the number of columns in the table
+        // >   * 100% minus the sum of the intrinsic percentage width of all prior columns in
+        // >     the table (further left when direction is "ltr" (right for "rtl"))
+        let mut total_intrinsic_percentage_width = 0.;
+        for column_index in 0..self.table.size.width {
+            let column_measure = &mut column_measures[column_index];
+            let final_intrinsic_percentage_width = column_measure
+                .percentage_width
+                .0
+                .min(100. - total_intrinsic_percentage_width);
+            total_intrinsic_percentage_width += final_intrinsic_percentage_width;
+            column_measure.percentage_width = Percentage(final_intrinsic_percentage_width);
+        }
+
+        self.column_measures = column_measures;
+    }
+
+    fn compute_content_sizes_for_columns_with_span_up_to_n(
+        &self,
+        n: usize,
+        old_column_measures: &[CellOrColumnMeasure],
+    ) -> (Option<usize>, Vec<CellOrColumnMeasure>) {
+        let mut next_span_n = None;
+        let mut new_content_sizes_for_columns = Vec::new();
+
+        for column_index in 0..self.table.size.width {
+            let old_column_measure = &old_column_measures[column_index];
+            let mut new_column_content_sizes = ContentSizes::zero();
+            let mut new_column_intrinsic_percentage_width = Percentage(0.);
+
+            for row_index in 0..self.table.size.height {
+                let coords = TableSlotCoordinates::new(column_index, row_index);
+                let resolved_coords = match self.table.resolve_first_cell_coords(coords) {
+                    Some(resolved_coords) => resolved_coords,
+                    None => continue,
+                };
+
+                let cell = match self.table.resolve_first_cell(resolved_coords) {
+                    Some(cell) if cell.colspan <= n => cell,
+                    Some(cell) => {
+                        next_span_n = next_span_n
+                            .map(|next_span_n: usize| next_span_n.min(cell.colspan))
+                            .or(Some(cell.colspan));
+                        continue;
+                    },
+                    _ => continue,
+                };
+
+                let cell_measures = &self.cell_measures[resolved_coords.y][resolved_coords.x];
+                let cell_inline_content_sizes = cell_measures.content_sizes;
+
+                let columns_spanned = resolved_coords.x..resolved_coords.x + cell.colspan;
+                let baseline_content_sizes: ContentSizes = (columns_spanned.clone())
+                    .map(|spanned_column_index| {
+                        old_column_measures[spanned_column_index].content_sizes
+                    })
+                    .fold(ContentSizes::zero(), |a: ContentSizes, b| a + b);
```

Comment:
Unnecessary parentheses around `columns_spanned.clone()`. Could also merge the map and the fold, but maybe it's less clear:

```suggestion
                let baseline_content_sizes: ContentSizes = columns_spanned.clone().fold(
                    ContentSizes::zero(),
                    |acc: ContentSizes, spanned_column_index| {
                        acc + old_column_measures[spanned_column_index].content_sizes
                    },
                );
```